src/lib/utils/retryImport.web.js

```diff
@@ -1,11 +1,27 @@
 // @flow
 import { noop } from 'lodash'
+import config from '../../config/config'
 import { retry } from './async'
+import restart from './restart'
+import { canReloadApp } from './canReloadApp'
+
+const unregisterWorkerAndRestart = async () => {
+  if (canReloadApp.current) {
+    const registrations = await navigator.serviceWorker.getRegistrations()
+
+    /* eslint-disable no-await-in-loop */
+    for (let registration of registrations) {
+      registration.scope.includes(config.publicUrl) && (await registration.unregister())
+    }
+
+    restart()
+  }
+}
 
 const retryImport = fn =>
   retry(fn, 5, 1000).catch(e => {
     const { name, message } = e
-
+    setTimeout(unregisterWorkerAndRestart, 30000)
```

Comment:
why timeout? why not after it fails to import 3 times?
also 30sec is too long, no user is going to wait that long.

Reply:
Maybe before timeout, modal (with message about update) will triggered with corresponding logic

Reply:
move that timeout **after** if statement. if we have other issue that "Unexpected token <" (error loading js chunk) we need just re-throw it, without reload

```suggestion
```

Reply:
let's decrease to 10-15sec

Reply:
@serdiukov-o-nordwhale i  still dont understand why we need timeout

Reply:
we have 2 cases
- issue happened just after app update. Here we're returning infinite promise, user is seeing the splash screen, then upgrade dialog appears. user dismissing it, the app reloads with the new version
- issue happened due to the other reason (e.g. very old version is cached, server issues etc). In this case we're waiting 10-15 sec for upgrade dialog. if no upgrade during that time period we're unregistering SW (to app could re-download the latest version and re-cache the manifest) and reloading